ocaml/runtime/major_gc.c

```diff
@@ -2042,9 +2056,14 @@ int caml_mark_stack_is_empty(void)
 }
 #endif
 
-void caml_empty_mark_stack (void)
+static void empty_mark_stack (void)
 {
   while (!Caml_state->marking_done){
+    while (!caml_marking_started()) {
```

Comment:
Add comment to justify this annoying-but-necessary while loop.

Reply:
Done